rust/osm2lanes/src/transform/tags_to_lanes.rs

```diff
@@ -499,83 +526,163 @@ fn parking(
     }
 }
 
-fn walking(
+fn foot_and_shoulder(
     tags: &Tags,
     locale: &Locale,
-    _oneway: bool,
+    oneway: bool,
     forward_side: &mut Vec<Lane>,
     backward_side: &mut Vec<Lane>,
-) {
-    if tags.is("sidewalk", "both") {
-        forward_side.push(Lane::foot());
-        backward_side.push(Lane::foot());
-    } else if tags.is("sidewalk", "separate") && locale.infer_sidewalks {
-        // TODO Need to snap separate sidewalks to ways. Until then, just do this.
-        forward_side.push(Lane::foot());
-        if !backward_side.is_empty() {
-            backward_side.push(Lane::foot());
+    warnings: &mut LaneWarnings,
+) -> ModeResult {
+    // https://wiki.openstreetmap.org/wiki/Key:sidewalk
```

Comment:
There were so many bizarre assumptions baked in here from old A/B Street design choices. Nice to see it done from scratch properly!